File: leet_1431_kids_with_most_candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        most = max(candies)
        res = []
        for n in candies:
            if n + extraCandies < most:
                res.append(False)
            else:
                res.append(True)
        return res

File: leet_1431_kids_with_most_candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies)
    {
        int most = *max_element(candies.begin(), candies.end());
        vector<bool>    res;

        for (int n : candies)
        {
            if (n + extraCandies < most)
                res.push_back(false);
            else
                res.push_back(true);
        }
        return res;
    }
};
